components/amount-buttons.jsx

```diff
@@ -1,50 +1,103 @@
 import React from 'react';
+import {IntlMixin, FormattedNumber} from 'react-intl';
+
+var AmountButton = React.createClass({
+  render: function() {
+    var checked = false;
+    if (this.props.value === this.props.amount) {
+      checked = true;
+    }
+
+    return (
+      <div className="third">
+        <input onChange={this.props.onChange} checked={checked} type="radio" name="donation_amount" value={this.props.value} id={"amount-" + this.props.value}/>
+        <label htmlFor={"amount-" + this.props.value} className="large-label-size">
+          <FormattedNumber
+            minimumFractionDigits={0}
+            value={this.props.value}
+            style="currency"
+            currency="USD"
+          />
+        </label>
+      </div>
+    );
+  }
+});
+
+var AmountOtherButton = React.createClass({
+  onRadioClick: function() {
+    document.querySelector("#amount-other-input").focus();
```

Comment:
Just a side note, better to use a `ref` here, or if there's some reason you can't,

```
this.getDOMNode().querySelector('...');
```